pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -197,16 +221,11 @@ func checkDiskSize(ctxPtr *zedmanagerContext) error {
 		appDiskSizeList += fmt.Sprintf("App: %s (Size: %d)\n%s\n",
 			iterStatus.UUIDandVersion.UUID.String(), appDiskSize, diskSizeList)
 	}
-	deviceDiskUsage, err := disk.Usage(types.PersistDir)
+	deviceDiskSize, diskReservedForDom0, err := diskSizeReservedForDom0(ctxPtr)
 	if err != nil {
-		err := fmt.Errorf("Failed to get diskUsage for /persist. err: %s",
-			err.Error())
-		log.Errorf("checkDiskSize: err:%s", err.Error())
+		log.Errorf("checkDiskSize: Failed to get maxDom0DiskSize. err: %s", err)
 		return err
 	}
-	deviceDiskSize := deviceDiskUsage.Total
-	diskReservedForDom0 := uint64(float64(deviceDiskSize) *
-		(float64(ctxPtr.globalConfig.GlobalValueInt(types.Dom0MinDiskUsagePercent)) * 0.01))
 	allowedDeviceDiskSizeForApps := deviceDiskSize - diskReservedForDom0
```

Comment:
Can the left hand side be negative? Is the type of allowedDeviceDiskSizeForApps a uint64 or an int64?
Seems like a check for negative would be useful.

Reply:
Hmm.. Good point.. Will add it.